extensions/api/data-management/contractagreement/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractagreement/ContractAgreementApiController.java

```diff
@@ -64,7 +66,14 @@ public List<ContractAgreementDto> getAllAgreements(@Valid @BeanParam QuerySpecDt
 
         monitor.debug(format("get all contract agreements from %s", spec));
 
-        return service.query(spec).stream()
+        var queryResult = service.query(spec);
+
+        //will throw an exception
```

Comment:
Instead of having to add this comment and having a method that throws exception, could the method create an exception from the parameters and then would be up to the client (in this case the controller) to throw it?

Reply:
that would be a bit cleaner, because the method wouldn't "secretly" throw the exception, but we would get more duplicated code.